src/relay/op/nn/convolution.h

```diff
@@ -153,6 +153,16 @@ bool Conv2DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
       << " But got " << out_layout;
 
   Array<IndexExpr> dshape_nchw = trans_in_layout.ForwardShape(data->shape);
+  bool is_depthwise = false;
+  if (param->groups > 1) {
+    CHECK(weight && weight->shape.defined()) <<
+        "Weight shape must be specified when groups is greater than 1.";
+    Array<IndexExpr> wshape_oihw = trans_kernel_layout.ForwardShape(weight->shape);
+    if (tvm::tir::Equal(param->groups, dshape_nchw[1]) &&
+        tvm::tir::Equal(param->groups, wshape_oihw[0])) {
```

Comment:
shouldn't it be `wshape_oihw[1]`? To me it looks like testing groups == in_channel == out_channel

Reply:
In depthwise conv2d, weight's out_channel == groups == input's in_channel, and weight's in_channel == channel multiplier. The output's channel = groups * channel multiplier.

Reply:
ah right, thanks